src/providers/BackgroundReplacementProvider/index.tsx

```diff
@@ -104,6 +111,18 @@ const BackgroundReplacementProvider: FC<Props> = ({
     };
   }, [replacementSpec, replacementOptions]);
 
+  useEffect(() => {
+    if (processor && observer) {
+      (processor as BackgroundReplacementProcessor).addObserver(observer);
+    }
+    
+    return () => {
+      if (processor && observer) {
+        (processor as BackgroundReplacementProcessor).removeObserver(observer);
+      }
+    };
+  }, [observer]);
```

Comment:
Should we add `processor` dependency as well?

Reply:
Yes, good catch. When a new processor is intitialized, we want to add the observers to that also.

Reply:
Probably missed adding here yet :)

Reply:
ops thanks

Reply:
That's actually a great catch